File: milica_and_string/milica.cpp
#include <iostream>
#include <algorithm>
#include <string>

using namespace std;

int main(void)
{
    int t;
    cin >> t;
    for (int i = 0; i < t; i++)
    {
        int n, k;
        cin >> n >> k;
        string s;
        cin >> s;
        int num_b = count(s.begin(), s.end(), 'B');
        if (num_b == k)
        {
            cout << 0 << "\n";
        }

        else if (num_b < k)
        {
            for (int j = 0; j < n; j++)
            {
                if (s[j] == 'B')
                {
                    continue;
                }

                s[j] = 'B';
                num_b++;
                if (num_b == k)
                {
                    cout << 1 << "\n" << j + 1 << " " << "B\n";
                }
            }
        }

        else
        {
            for (int j = 0; j < n; j++)
            {
                if (s[j] == 'A')
                {
                    continue;
                }

                s[j] = 'A';
                num_b--;
                if (num_b == k)
                {
                    cout << 1 << "\n" << j + 1 << " " << "A\n";
                }
            }
        }
    }
}

File: milica_and_string/milica.py
def main():
    t = int(input())
    for _ in range(t):
        n, k = map(int, input().split())
        s = list(input())
        num_b = s.count('B')
        if (num_b == k):
            print(0)
        elif (num_b < k):
            for i in range(n):
                if s[i] == 'B':
                    continue
                else:
                    s[i] = 'B'
                    num_b += 1
                    if (num_b == k):
                        ans = i + 1
                        break
            print(1)
            print(ans, 'B')
        else:
            for i in range(n):
                if s[i] == 'A':
                    continue
                else:
                    s[i] = 'A'
                    num_b -= 1
                    if (num_b == k):
                        ans = i + 1
                        break
            print(1)
            print(ans, 'A')
            
    
    
if __name__ == "__main__":
    main()
